src/main/kotlin/no/nav/familie/ef/sak/vurdering/medlemskap/MedlemskapRegelsett.kt

```diff
@@ -0,0 +1,246 @@
+
+    private val oppholdsrett = (nordiskStatsborger eller eøsBorgerEllerLovligOpphold)
+            .med("FP_VK 2.12", "Har søker lovlig opphold i Norge eller er søker EØS-borger med oppholdsrett")
+
+    private val forutgåendeMedlemskapSøker =
+            (medlemskapslengde eller medlemskapslengdeAvbruddMindreEnn10år eller medlemskapslengdeAvbruddMerEnn10år)
+                    .med("EF-M7 EF-M8 EF-M10", "Er vilkår for forutgående medlemskap oppfylt?")
+
+    val vurderingMedlemskapSøker =
+            (erMedlem
+                    og erBosattINorge
+                    og (oppholdINorge eller utenlandsoppholdGrunnetNorskArbeidsgiver)
+                    og forutgåendeMedlemskapSøker
+                    og oppholdsrett).med("Medlemskap kap. 15",
+                                         "Har søker opphold i Norge og forutgående medlemskap?")
+
+    private fun erMedlem(medlemskapsgrunnlag: Medlemskapsgrunnlag): Evaluering {
+        val statusNå =
+                medlemskapsgrunnlag.medlemskapshistorikk.medlemskapsperioder.find { it.inneholder(LocalDate.now()) }
+        if (statusNå == null || statusNå.gyldig == false) {
+            return Evaluering(Resultat.NEI, "Er IKKE MEDLEM")
+        }
+        if (statusNå.gyldig == null) {
+            return Evaluering(Resultat.KANSKJE, "Uavklart medlemskapsunntak nå")
+        }
+        return Evaluering.ja("Søke er registrert som medlem nå.")
+    }
+
+    private fun erBosattINorge(medlemskapsgrunnlag: Medlemskapsgrunnlag): Evaluering {
+        return if ("bosatt" == medlemskapsgrunnlag.søker.person.folkeregisterpersonstatus.firstOrNull()?.status) {
+            Evaluering(Resultat.JA, "Bosatt i Norge.")
+        } else {
+            Evaluering(Resultat.NEI, "Ikke bosatt i Norge.")
+        }
+    }
+
+    private fun oppholdINorge(medlemskapsgrunnlag: Medlemskapsgrunnlag): Evaluering {
+        return if (medlemskapsgrunnlag.søknad.medlemskapsdetaljer.verdi.oppholderDuDegINorge.verdi) {
+            Evaluering(Resultat.JA, "Oppholder seg i Norge.")
+        } else {
+            Evaluering(Resultat.NEI, "Oppholder seg ikke i Norge.")
+        }
+    }
+
+    private fun utenlandsoppholdGrunnetNorskArbeidsgiver(medlemskapsgrunnlag: Medlemskapsgrunnlag): Evaluering {
+        // Hvis dette blir en del av søknaden kan vi gi bedre svar.
+        return Evaluering(Resultat.KANSKJE, "Saksbehandler må ta stilling til om utenlandsopphold grunnes arbeid" +
+                                            "for norsk arbeidsgiver.")
+    }
+
+
+    private fun medlemskapslengde(medlemskapsgrunnlag: Medlemskapsgrunnlag): Evaluering {
+        val søknadsdato = søknadsdato(medlemskapsgrunnlag)
+
+        val statusSisteÅr = medlemskapsgrunnlag.medlemskapshistorikk.medlemskapsperioder.filter {
+            it.tildato.isAfter(søknadsdato.minusYears(antallÅrOpphold))
+        }
+
+        if (statusSisteÅr.find { it.gyldig == false } != null) {
```

Comment:
Her(og noen andre steder) er det mulig å bruke`.any` istedet for `.find() != null`